### PR TITLE
Update Swift from 3.0 to 4.2

### PR DESCRIPTION
**Pull request by yassinecc**
Xcode 10.2 is not compatible with Swift 3.0

### CODE REVIEW

**Comment by yassinecc** [comment]
@maartenschumacher Yes build is fine on simulator, as well as pushing / popping screens. Not sure of what extra tests I can run, maybe push notifications or deep links?

**Comment by Almouro** [comment]
We should basically check that was modified still works (e.g. setting the title...)

A bit weirded out by the `convert` functions added, we should check if we could get rid of those somehow

File: lib/ios/native-navigation/ReactViewControllerProtocol.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ extension ReactTabBarController: InternalReactViewControllerProtocol {
 // we will wait a maximum of 200ms for the RN view to tell us what the navigation bar should look like.
 // should normally happen much quicker than this... This is just to make sure it transitions in a reasonable
 // time frame even if the react thread takes an extra long time.
-private let DELAY: Int64 = Int64(1.2 * Double(NSEC_PER_SEC))
+public let DELAY: Int64 = Int64(1.2 * Double(NSEC_PER_SEC))
```
Comment (Almouro) [inline lib/ios/native-navigation/ReactViewControllerProtocol.swift]: I think this could be added in a separate constant file for instance

**Pull request merged**